benchmarks/dbscan/data.hpp

```diff
@@ -26,19 +26,22 @@ std::vector<Point<DIM>> sampleData(std::vector<Point<DIM>> const &data,
 {
   std::vector<Point<DIM>> sampled_data(num_samples);
 
-  std::default_random_engine generator(17);
-  // FIXME_NVCC(11.0-11.3): not providing int here results in an internal
-  // compiler error
-  std::uniform_int_distribution<int> distribution;
-  auto rand = [&distribution, &generator]() { return distribution(generator); };
+  // Lehmer (or Park-Miller) RNG
+  assert(num_samples > 1);
+  unsigned int state =
+      1337 % (num_samples - 1) + 1; // any positive number less than modulus
```

Comment:
Why would you want to expose the state outside the lambda?
Make it a static variable within the lambda's local scope.